test/utils/gtest_formating_checks.py

```diff
@@ -0,0 +1,89 @@
+import os
+import re
+import logging
+
+logging.basicConfig(level=logging.INFO, format='%(levelname)s: %(message)s')
+
+FOLDER_PATH = "../../test/gtest"
+
+# Ignore list: Add test names or file paths you want to exclude
+IGNORE_LIST = {
+    "CPU_MIOpenDriverRegressionBigTensorTest_FP32",
+    "../../test/gtest/reduce_custom_fp32.cpp",
+}
+
+# Valid enums and Regex for validation
+VALID_HW_TYPES = {"CPU", "GPU"}
+VALID_DATATYPES = {"FP8", "FP16", "FP32", "FP64", "BFP16", "BFP8", "I64", "I32", "I16", "I8", "NONE"}
+TESTSUITE_REGEX = re.compile(
+    r"^(CPU|GPU)_[A-Za-z0-9]+(?:_[A-Za-z0-9]+)*_(" + "|".join(VALID_DATATYPES) + r")$"
+)
```

Comment:
Doesn't it duplicate https://github.com/ROCm/MIOpen/pull/3218 ?

Reply:
This one extends on it + CI integration that runs upon adding new gtest

Reply:
Do we need separate python scripts?
Seems like we might be able to unify these.